### PR TITLE
Feat/jest setup

### PR DESCRIPTION
**Pull request by nRamstedt**
- Installed React Native Testing Library
- Added custom render method (test-utils.js) to allow re-using wrapper components across tests
- Added MockNavigator for testning components that depend on Navigator API
- Added example integration test of Form Component
- Added integration test of LoginScreen
- Mock native modules: react-native-config, react-native-gesture-handler, react-native-reanimated etc (Within jestSetupFile.js)

To run tests:
1. yarn install
2. npx jest --coverage

### CODE REVIEW

File: jestSetupFile.js [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+
+/**
+ * Mock react-native-image-picker
+ */
+NativeModules.ImagePickerManager = {
+    showImagePicker: jest.fn(),
+    launchCamera: jest.fn(),
+    launchImageLibrary: jest.fn(),
+};
+
```
Comment (zilaei) [inline jestSetupFile.js:29]: Getting prettier errors for line indentation.

File: test-utils.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { AuthProvider } from 'app/store/AuthContext';
+import ScreenWrapper from 'app/components/molecules/ScreenWrapper';
+
+const RenderWrapper = ({children}) => (
+    <AuthProvider>
+        <ScreenWrapper>
+            {children}
+        </ScreenWrapper>
+    </AuthProvider>
+);
```
Comment (zilaei) [inline test-utils.js:12]: Getting prettier errors for line indentation.

File: test-utils.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import { AuthProvider } from 'app/store/AuthContext';
+import ScreenWrapper from 'app/components/molecules/ScreenWrapper';
+
+const RenderWrapper = ({children}) => (
```
Comment (zilaei) [inline test-utils.js:6]: Missing prop validation for children.

File: test-utils.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    <AuthProvider>
+        <ScreenWrapper>
+            {children}
+        </ScreenWrapper>
+    </AuthProvider>
+);
+  
+
+const customRender = (ui, options) => render(ui, { wrapper: RenderWrapper });
```
Comment (zilaei) [inline test-utils.js:15]: Unused parameter options.

**Review by zilaei — COMMENTED** [review]
Getting some prettier warnings for some of the files.

File: __tests__/Login.test.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+mock.onPost('/auth/token?apikey=').reply(200, {
+  success: true,
+  data: {
+    attributes: {
+      token: '123',
+    },
+  },
+});
+
```
Comment (TeddyAndersson) [inline __tests__/Login.test.js:56]: Maybe we also need a test for starting the login process? ie when a user starts the login and the bankid loading screens open? Or should that test be placed somewhere else?

File: __tests__/Login.test.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      completion_data: {
+        user: {
+          personal_number: '199803312389',
+        },
+      },
+    },
+  },
+});
+
+mock.onPost('/auth/token?apikey=').reply(200, {
```
[the remark this answers is not in the document]
Comment (nRamstedt) [reply __tests__/Login.test.js:48]: This i probably out of this PR scope

**Pull request merged**